test/acl_device_test.cpp

```diff
@@ -327,6 +327,9 @@ MT_TEST(DeviceInfo, basic) {
     if (queries[i] != CL_DEVICE_EXTENSIONS) {
       // Non-empty result for most supported queries.
       CHECK(size_ret > 0);
+    } else {
+      // Check buffer location is a supported extension
+      CHECK(strstr(str, "cl_intel_mem_alloc_buffer_location"));
     }
```

Comment:
This check is unrelated to the `size_ret`, so I would suggest adding a separate section in the same style as the other query tests, i.e. `if (queries[i] == CL_DEVICE_EXTENSIONS) { ... }`.

This only tests for this particular extension, would you like to test for the other extensions too? Instead of hard-coding the extensions string, you could also compare for equality with `m_platform->extensions`, to avoid duplicating the list of extensions, e.g., as in [this test in the GPU runtime](https://github.com/intel/compute-runtime/blob/fc2f8cadfd450aa3071b9ef67b76edaadfd335bc/opencl/test/unit_test/device/device_caps_tests.cpp#L967). To ensure the test is robust (what if `m_platform->extensions` is corrupted for some reason?), you could factor out the extension string into a `#define` and use that to both initialize `m_platform->extensions` and to compare with in the unit test.

Reply:
I have pushed few commits

the first on added the extension name and preliminary test
the second remove if statement check around the return size
the third added full checks for all extension.

Details are in the commit message.